src/components/views/EditMilestone.jsx

```diff
@@ -128,13 +145,46 @@ class EditMilestone extends Component {
               );
             });
         } else {
+          const qs = queryString.parse(this.props.location.search);
           CampaignService.get(this.props.match.params.id)
             .then(campaign => {
               if (campaign.projectId < 0) {
                 this.props.history.goBack();
               } else {
                 const { milestone } = this.state;
-                milestone.recipientAddress = this.props.currentUser.address;
+                validQueryStringVariables.forEach(variable => {
+                  if (qs[variable]) {
+                    if (variable === 'fiatAmount') {
+                      milestone.fiatAmount = new BigNumber(qs[variable]);
```

Comment:
There should be a check for what was passed. If I pass do `fiatAmount=` it will throw and the website will just stay "Unable to load milestone".

Reply:
Same applies to other variables.